packages/shared/lib/core/network/interfaces/auth.interface.ts

```diff
@@ -0,0 +1,5 @@
+export interface IAuth {
```

Comment:
I think we should leave this as `NodeAuth` rather than just `Auth` since it is specifically for client options / node connections. 

Also, any shortened names should use the full version so `NodeAuthentication` rather than just `NodeAuth` here.

Reply:
Ahh but we had really weird names because of this, like getNodeAuth and getNodeAuthNode. This has no mention of node in it and could be reused for any auth object in the future. So renamed to getAuth and getNodeAuth.

> Also, any shortened names should use the full version so NodeAuthentication rather than just NodeAuth here
Again just copying code here, I generally agree with the statement. But for this specific case I would argue Auth is an acceptable shortening and used pretty commonly.

Reply:
> * `utils` could be renamed `helpers` as that's a slightly more accurate description of the code that is there.

This makes sense in some cases, I think helpers should only be used internally in a module and utils can be used in the components. Lets make this change going forwards when I refactor the networkStatus lib.

> * There should be TSDoc comments for all code components, i.e. functions, stores, types / interfaces, constants (I have yet to write conventions for how we should write these comments to make them read better and more consistently... will work on this soon).

Agreed, although I didn't write these functions and I have modified any that existed so that they are correct. We can slowly add them over time but our priority isn't currently commenting existing code. 

> * We should be using regular functions declared with the `function` keyword rather than ES6-type arrow functions (see [here](https://iotaledger.github.io/firefly/guides/coding-conventions/typescript-library/#functions)).

Agreed, I tried to migrate as many as possible when moving them over, is there specific ones that I missed?

> * We should use `interface`s primarily instead of `type` unless absolutely necessary, however IMO we shouldn't prefix the names with "I" unless it is an interface specifically meant to be implemented somewhere else. I think the "I" is unnecessary whenever it's used as a normal type as it doesn't really indicate anything extra of value to us.

Lol this will be a long debate, I think its hard to be objective on something that are both pretty much the same. I do think that the the I helps identifying. How about we use interfaces for object literals and types for everything else?

Reply:
> Ahh but we had really weird names because of this, like getNodeAuth and getNodeAuthNode. This has no mention of node in it and could be reused for any auth object in the future. So renamed to getAuth and getNodeAuth.

Yeah that's true, except `getAuth` and `getNodeAuth` sound like they would return the same thing. Maybe `getAuth` and `getNodeWithAuth` would be better?

> But for this specific case I would argue Auth is an acceptable shortening and used pretty commonly.

True, let's keep it :+1: